### ISSUE TITLE
Move from Azure Table Storage to  Cosmos DB

### ISSUE THREAD

**Issue by olemp** [Puzzlepart/did#815]
- [x] I have searched for existing issues to avoid creating duplicates

### TLDR
Move from Azure Table Storage to Cosmos DB.

### Decisions
- [ ] How to handle BYOS (bring your own storage) with `mongodb`. The prefered way is a database per customer.

### Description
[The Azure Cosmos DB Table API is nearly identical to Azure Table Storage](https://docs.microsoft.com/en-us/azure/cosmos-db/table-support), but provides higher throughput and scalability.

The [azure-storage](https://www.npmjs.com/package/azure-storage) SDK that we're already using for Azure Table Storage can also be used with the Cosmos DB Table API ([source](https://docs.microsoft.com/en-us/azure/cosmos-db/table-support#developing-with-the-azure-cosmos-db-table-api)), using the same CRUD-operations.

Many of the bulk-operations we're doing today, e.g. importing and exporting TimeEntries from one Table Storage table to another (~26k entries, 23mb), are already appearing cumbersome and "too big".

### Related issues
See #506 for discussion.
